src/components/views/admin/BuilderView.tsx

```diff
@@ -1,156 +1,243 @@
+    { value: 'referencesInfo', label: 'References Info' },
+    { value: 'cvStatusInfo', label: '' },
+  ];
+
+  // Construct the 'cv' object with filtered and grouped data
+  const candidatesCv: { [key: string]: { [key: string]: any } } = {};
+
+  dimensions.forEach((dimension) => {
+    const type = dimension.value;
+    const dataObj = Object.keys(groupedData)
+      .filter((key) => key === type)
+      .map((key) => groupedData[key]);
+
+    const typeDataByMemberId = DataFilter({
+      dataObject: dataObj,
+      targetType: type,
+    });
+
+    Object.keys(typeDataByMemberId).forEach((memberId) => {
+      if (memberId in candidatesCv) {
+        candidatesCv[memberId][type] = typeDataByMemberId[memberId].map(
+          (item) => item.data,
+        );
+      } else {
+        candidatesCv[memberId] = {
+          [type]: typeDataByMemberId[memberId].map((item) => item.data),
+        };
+      }
+    });
+  });
+
+  const [ratings, setRatings] = useState<{ [key: string]: number | null }>({});
+  const handleRatingChange = (
+    memberId: string,
+    newRating: number | null,
+  ): void => {
+    const rateData = appDataArray.find(
+      (data) =>
+        data.data.memberId === memberId && data.type === 'candidateRateInfo',
+    );
+    if (rateData) {
+      handlePatch(rateData.id, {
+        ...rateData.data,
+        rating: newRating,
+        memberId,
+      });
+    } else {
+      handlePost({ memberId, rating: newRating });
+    }
+    setRatings((prevRatings) => ({
+      ...prevRatings,
+      [memberId]: newRating,
+    }));
+  };
+
+  const handleDownload = async (memberId: string): Promise<void> => {
+    const candidateCvData = candidatesCv[memberId];
+    const { component: CvTemplate } = TEMPLATES.find(
+      (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+    ) || { component: FirstTemplate };
+    const renderedTemplate = <CvTemplate cvValues={candidateCvData} />;
+    const pdfBlob = await pdf(renderedTemplate).toBlob();
+    if (!candidateCvData.cvStatusInfo.customCv) {
+      saveAs(pdfBlob, 'generated-cv.pdf');
+    }
+  };
+  const [visibility, setVisibility] = useState<string | null>(null);
+  const [renderedFile, setRenderedFile] = useState<string | null>(null);
+  const handleView = async (memberId: string): Promise<void> => {
+    if (visibility === memberId) {
+      setVisibility(null);
+      setRenderedFile(null);
+    } else {
+      const candidateCvData = candidatesCv[memberId];
+      const { component: CvTemplate } = TEMPLATES.find(
+        (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+      ) || { component: FirstTemplate };
+      const fileToRender = <CvTemplate cvValues={candidateCvData} />;
+      const pdfBlob = await pdf(fileToRender).toBlob();
+      const dataUrl = URL.createObjectURL(pdfBlob);
+      setVisibility(memberId);
+      setRenderedFile(dataUrl);
+      window.open(dataUrl, '_blank');
```

Comment:
I did not think about opening the PDFs in a new tab. Wouldn't this break the flow of using the app in Graasp, from a UX point of view ?

Reply:
Yeh you are right, but it would be a little confusing for maybe the Reviewer to where the file would be shown, and a new tab is a good option where they would just see the desired file, what do you think?